src/modules/InstURL.rb

```diff
@@ -86,24 +87,27 @@ def RewriteCDUrl(url)
     def installInf2Url(extra_dir = "")
       return @installInf2Url unless @installInf2Url.nil?
 
-      parts = URL.Parse(Linuxrc.InstallInf("ZyppRepoURL"))
+      zypp_repo_url = Linuxrc.InstallInf("ZyppRepoURL").to_s
 
-      if parts.empty?
+      if zypp_repo_url.empty?
         # Make it compatible with the current behaviour when
         # install.inf does not exist.
         log.warn "No URL specified through ZyppRepoURL"
-        parts = URL.Parse("cd:///")
+        zypp_repo_url = "cd:/"
       end
 
-      parts["path"] = File.join(parts["path"], extra_dir) unless extra_dir.empty?
+      uri = URI(zypp_repo_url)
```

Comment:
The proposal LGTM.

I know that working with the parsed URL is simpler and can be easily extended, but libzypp sometimes uses quite complicated URLs (`iso://...`, `smb://...`) and the risk of building the URL back wrongly is quite high. In this case I'd rather stay on the safe side and sacrifice the better code readability.